File: java/1646_GetMaximumInGeneratedArray.java
class Solution {
    public int getMaximumGenerated(int n) {
        if (n <= 0) return 0;
        
        int[] dp = new int[n+1];
        dp[1] = 1;
        
        int high = 0;
        for (int i = 0; i <= n; ++i) {
            dp[i] = dp[i / 2] + (i % 2 == 1 ? dp[i / 2 + 1] : 0);
            high = Math.max(high, dp[i]);
        }
        return high;
    }
}

File: swift/1646_GetMaximumInGeneratedArray.swift
class Solution {
    func getMaximumGenerated(_ n: Int) -> Int {
        if n <= 0 { return 0 }
        var dp = [Int](repeating: 0, count: n + 1)
        dp[1] = 1
        var high = 1
        
        for i in 0...n {
            dp[i] = i % 2 == 0 ? dp[i / 2] : dp[i / 2] + dp[i / 2] + 1
            high = max(high, dp[i])
        }
        return high
    }
}
